packages/commercetools/theme/pages/Checkout.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <div id="checkout">
+    <div class="checkout">
+      <div class="checkout__main">
+        <SfSteps :active="currentStepIndex" @change="handleStepClick" v-if="!isThankYou" :class="{ 'checkout__steps': true, 'checkout__steps-auth': isAuthenticated }">
+          <SfStep v-for="(step, key) in STEPS" :key="key" :name="step">
+            <nuxt-child />
+          </SfStep>
+        </SfSteps>
+        <nuxt-child v-else  />
+      </div>
+      <div class="checkout__aside desktop-only" v-if="!isThankYou">
+        <transition name="fade">
+          <CartPreview v-if="showCartPreview" key="order-summary" />
+          <OrderReview v-else key="order-review" />
+        </transition>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+
+import { SfSteps, SfButton } from '@storefront-ui/vue';
+import CartPreview from '~/components/checkout/CartPreview';
+import OrderReview from '~/components/checkout/OrderReview';
+import { ref, computed } from '@vue/composition-api';
+import { useUser } from '@vue-storefront/commercetools';
+
+const STEPS = {
+  'personal-details': 'Personal Details',
+  shipping: 'Shipping',
+  payment: 'Payment',
+  'order-review': 'Review'
+};
+
+export default {
+  name: 'Checkout',
+  components: {
+    SfButton,
+    SfSteps,
+    CartPreview,
+    OrderReview
+  },
+  setup(props, context) {
+    const currentStep = computed(() =>context.root.$route.path.split('/').pop());
+    const { isAuthenticated } = useUser();
+    const showCartPreview = ref(true);
+    const currentStepIndex = computed(() => Object.keys(STEPS).findIndex(s => s === currentStep.value));
+    const isThankYou = computed(() => currentStep.value === 'thank-you');
+
+    const handleStepClick = (stepIndex) => {
+      const key = Object.keys(STEPS)[stepIndex];
```

Comment:
`key` can be possibly undefined. Can you add some default value for prevention?

Reply:
it will always be a real index